messaging/src/main/java/org/axonframework/eventhandling/GenericTrackedSourcedEventMessage.java

```diff
@@ -0,0 +1,31 @@
+package org.axonframework.eventhandling;
+
+/**
+ * Generic implementation of a {@link TrackedSourcedEventMessage}.
+ *
+ * @param <T> The type of payload contained in this Message
+ */
+public class GenericTrackedSourcedEventMessage<T> extends GenericTrackedEventMessage<T>
```

Comment:
The `source` is conceptually strongly tied with the `TrackedEventMessage` (or more likely, just the `StreamableMessageSource`). A `token`, for example, can only be compared to another token if they have been received through streams from the same source.
Instead of adding a layer of indirection, consider adding the source to existing types.

Reply:
From reading this comment and this class, I am still uncertain whether this comment has been resolved. Has it?